### PR TITLE
feat: simulation plan playground

### PR DESCRIPTION
**Pull request by mwdchang**
## Description
This adds basic renderer and parser for a simulation-plans, note this is quite rough as the simulation format itself is currently underspecified (there is insufficient information to determine the appropriate user interactions). 

The purpose here is to have some code and utilities in place that we can copy from later.

## Testing
Navigate to http://localhost:8078/app/#/simulation-plan-playground, you should see a plan for doing average-treatment-effects.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### CODE REVIEW

**Review by YohannParis — DISMISSED** [review]
Looks good and works for me.

File: packages/client/hmi-client/src/services/simulation.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+	OuterOutPort: { outer_out_port_type: string }[];
+	OuterInPort: { outer_in_port_type: string }[];
+}
+
+/**
+ * Given a simulation plan wiring diagram, convert into a node/edges graph
+ * format.
+ *
+ * Note currently this only processes: "InPort", "OutPort", "Wire", "Box"
```
[the remark this answers is not in the document]
Comment (mwdchang) [reply packages/client/hmi-client/src/services/simulation.ts:25]: There aren't any documentations, I still don't understand what pass-wire is despite asking TA-2 folks. But relation-wise it is parallel array type of thing with indexing.

**Review by bencodrington — APPROVED** [review]
Looks good, works for me too. Thanks

**Review by bencodrington — APPROVED** [review]

File: packages/client/hmi-client/src/services/simulation.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (YohannParis) [reply packages/client/hmi-client/src/services/simulation.ts:25]: @jpfairbanks do you know where we could find some information about those terms?

**Pull request merged**